sklearn/inspection/_partial_dependence.py

```diff
@@ -25,14 +26,16 @@
 from ..utils import _determine_key_type
 from ..utils import _get_column_indices
 from ..utils.validation import check_is_fitted
-from ..tree._tree import DTYPE
 from ..exceptions import NotFittedError
 from ..ensemble._gb import BaseGradientBoosting
 from sklearn.ensemble._hist_gradient_boosting.gradient_boosting import (
     BaseHistGradientBoosting)
 
-
-__all__ = ['partial_dependence', 'plot_partial_dependence',
+__all__ = ['individual_conditional_expectation',
```

Comment:
We have 1500 lines of code now. I think this the moment to split things.

I would create a submodule called `_plot` and add the plotting facilities there.
We should also rename the file from `_partial_dependence` to something which is both working for PD and ICE.

Basically, we need a nice name for explaining that we deal with global inspection which check the marginal effect of a feature(s) with the target. So I don't know if we should call it `_global.py`. Everything that we have for the moment is indeed a global inspection. Maybe `_feature_interaction.py` might be better.